index.js

```diff
@@ -196,9 +218,14 @@ module.exports = postcss.plugin('postcss-sorting', function (opts) {
 
                     var prevNode = node.prev();
 
-                    if (prevNode && node.groupIndex > prevNode.groupIndex) {
-                        if (node.raws.before) {
-                            node.raws.before = '\n' + node.raws.before;
+                    if (prevNode && node.raws.before) {
+                        if (node.groupIndex > prevNode.groupIndex) {
+                            node.raws.before = createLineBreaks(1) + node.raws.before;
+                        }
+
+                        // Insert empty lines between children classes
+                        if(node.sortName === '>child' && linesBetweenChildrenRules > 0 && prevNode.groupIndex === node.groupIndex) {
```

Comment:
Let's use:

```js
if (node.type === 'rule' && prevNode.type === 'rule' && linesBetweenChildrenRules > 0) {
```

`prevNode.groupIndex === node.groupIndex` will be true even if previous node is not child. Like in the test you've added `background-image` and `&__row {}` have same groupIndex and empty line will be inserted but they shouldn't.